### ISSUE TITLE
grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC

### ISSUE THREAD

**Issue by A-Jabbar** [dydxprotocol/v4-clients#99]
Getting the following error while placing an error using Composite client example.

```
rpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Stream removed"
        debug_error_string = "UNKNOWN:Error received from peer  {created_time:"2023-12-21T08:49:46.070181+00:00", grpc_status:2, grpc_message:"Stream removed"}"
>
```

What could be the issue?

**Comment by ahdorll** [comment]
I'm getting the same error while executing examples/composite_example.py

**Comment by 0xCryptoLight** [comment]
See this issue: https://github.com/dydxprotocol/v4-clients/issues/81
Doesn't look like mainnet is properly configured in the repo.

Try instantiating CompositeClient as so:
`    compositeClient = CompositeClient(
        network=Network(
            env='mainnet',
            validator_config=ValidatorConfig(
                grpc_endpoint=VALIDATOR_GRPC_MAINNET, # use an RPC endpoint from https://docs.dydx.trade/networks/network1/resources
                chain_id='dydx-mainnet-1', 
                ssl_enabled=True
            ),
            indexer_config=indexerConfig,
            faucet_endpoint=None,
        )
    )`
    
  Doing that led me to this error:
```
  grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
        status = StatusCode.UNAVAILABLE
        details = "DNS resolution failed for ----omitted---: C-ares status is not ARES_SUCCESS qtype=A name=https is_balancer=0: Domain name not found"
        debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"DNS resolution failed for ----omitted----: C-ares status is not ARES_SUCCESS qtype=A name=https is_balancer=0: Domain name not found", grpc_status:14, created_time:"----omitted----"}"
```

**Comment by A-Jabbar** [comment]
I'm experiencing this error on the testnet even. Haven't moved to the mainnet part yet.

**Comment by 0xCryptoLight** [comment]
> I'm experiencing this error on the testnet even. Haven't moved to the mainnet part yet.

According to this [comment](https://github.com/dydxprotocol/v4-clients/issues/86#issuecomment-1840529642) the testnet RPCs are discontinued so you'd have to set up your own for testnet or use mainnet.

**Comment by A-Jabbar** [comment]
Ah, thanks. Have you been able to run on the mainnet through the workaround you mentioned in the previous comment? As I didn't find any mainnet in the Network.

**Comment by 0xCryptoLight** [comment]
> Ah, thanks. Have you been able to run on the mainnet through the workaround you mentioned in the previous comment? As I didn't find any mainnet in the Network.
Yes I was able to find the endpoints in the documentation. My issue was that I was using the RPC instead of the gRPC. Take a look here https://github.com/dydxprotocol/v4-clients/issues/102

**Comment by ahdorll** [comment]
in my case, the problem is resolved by fixing the line

v4-client-py/v4_client_py/clients/modules/post.py line 46
`network = NetworkConfig.fetch_dydx_testnet()`

to 
` network = NetworkConfig(
          chain_id="dydx-mainnet-1",
          url="grpc+https://dydx-grpc.publicnode.com:443",
          fee_minimum_gas_price=4630550000000000,
          fee_denomination="adv4tnt",
          staking_denomination="dv4tnt",
          faucet_url=None,
      )`

I think the fetch_dydx_testnet() is sending the call to testnet despite setting the mainnet configuration

**Comment by vincent7q** [comment]
there is a bug in v4-client-py/v4_client_py/clients/modules/post.py line 46 to get testnet network config. Below change can fix the problem

change source code as following (post.py)
      #network = NetworkConfig.fetch_dydx_testnet()       #bug to get testnet network config
        network= NetworkConfig(                             
            chain_id=self.config.chain_id, 
            url=self.config.grpc_endpoint, 
            fee_minimum_gas_price=0,    
            fee_denomination="adydx",   
            staking_denomination="adydx",  
            faucet_url=None, )

**Comment by A-Jabbar** [comment]
Managed to place the limit order on the mainnet by making the following change in the v4-client-py/v4_client_py/clients/modules/post.py line 46

`network = NetworkConfig.fetch_dydx_testnet() #bug to get testnet network config`

changed to 
`network = NetworkConfig( chain_id="dydx-mainnet-1", url="grpc+https://dydx-grpc.publicnode.com:443", fee_minimum_gas_price=0, fee_denomination="adydx", staking_denomination="adydx", faucet_url=None)`


However, while trying orderr type "MARKET" with IOC, facing the same issue while fetching the latest block using validator_client. Does anyone know the correct mainnet URLs for the validator and indexer clients?

**Comment by A-Jabbar** [comment]
Found the validator and indexer configs for the mainnet here #81 

```
const indexerConfig = new IndexerConfig(
'https://indexer.dydx.trade/',
'wss://indexer.dydx.trade/v4/ws'
);

const denomConfig = {
USDC_DENOM: 'ibc/8E27BA2D5493AF5636760E354E46004562C46AB7EC0CC4C1CA14E9E20E2545B5',
USDC_DECIMALS: 6,
USDC_GAS_DENOM: 'uusdc',
CHAINTOKEN_DENOM: 'adydx', //string
CHAINTOKEN_DECIMALS: 18 //integer
};

const validatorConfig = new ValidatorConfig(
'https://dydx-ops-rpc.kingnodes.com/',
'dydx-mainnet-1',
denomConfig
);

const custom_network = new Network(
'custom-network-name',
indexerConfig,
validatorConfig
);

_indexClient = new IndexerClient(custom_network.indexerConfig);
_validatorClient = await ValidatorClient.connect(custom_network.validatorConfig);
_compositeClient = await CompositeClient.connect(custom_network);
```
